tools/autograd/gen_python_functions.py

```diff
@@ -381,9 +388,13 @@ def get_simple_return_type(declaration):
 def get_dispatch_callee(declaration):
     # format the name of the receiving function or method
     if is_tensor_method(declaration):
+        if TOUtils.check_if_factory_method(declaration['arguments']):
+            return 'self._{}'.format(declaration['name'])
```

Comment:
I remember we talked about whether or not the exploded TensorOptions binding in `native_functions.yaml` should be underscored or not. The special case here seems like evidence that it *should* be underscored, because otherwise you don't have a consistent calling convention and have to do another special case here. It would be even better if the expanded variant was just called `empty`, and then Tensor is overloaded to also accept just a `TensorOptions`. If I recall correctly, you told me that the overload there would be ambiguous. But I don't really see why; the two calling conventions have differing number of arguments.

Reply:
I think this comment is mostly correct.

The thinking was, if we couldn't solve the ambiguous overload problem, it is strictly better to have a simple "call this other function" annotation rather than making each binding handle TensorOptions.

If we could solve the ambiguous overload problem, the question was whether the exploaded version should be "public", e.g. if we make that be the default call, people might call it instead of the TensorOptions variant (both because it's the function they were already calling, and it's on the surface simpler than figuring out what TensorOptions is).